plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/data/SQLQueryDataContainer.java

```diff
@@ -100,8 +100,8 @@ public DBCStatistics readData(
             SQLSyntaxManager syntaxManager = new SQLSyntaxManager();
             syntaxManager.init(dataSource.getSQLDialect(), dataSource.getContainer().getPreferenceStore());
             SQLRuleManager ruleManager = new SQLRuleManager(syntaxManager);
-            ruleManager.loadRules(dataSource, false);
-            SQLParserContext parserContext = new SQLParserContext(getDataSource(), syntaxManager, ruleManager, new Document(query.getText()));
+            ruleManager.loadRules(getDataSourceContainer(), false);
+            SQLParserContext parserContext = new SQLParserContext(getDataSourceContainer(), syntaxManager, ruleManager, new Document(query.getText()));
```

Comment:
I don't understand this approach at all. Why do you change dataSource to dataSourceContainer everywhere? Is it only for cases where we don't have dataSource?

Reply:
Yes, probably it seems more right because we can extract dialect by dataSourceConnection without even connecting